### PR TITLE
[Pipelines] Add support for external rabbitmq and redis

### PR DESCRIPTION
**Pull request by vivekkumar-git**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Chart Version bumped
- [ ] CHANGELOG.md updated
- [ ] Variables and other changes are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/pipelines/values.yaml [hunk 1]
```diff
@@ -502,7 +502,7 @@ pipelines:
         defaultExchange: {{ .Values.rabbitmq.root_vhost_exchange_name }}
         amqpVhost: {{ .Values.rabbitmq.build_vhost_name}}
         amqpRootVhost: {{ .Values.rabbitmq.root_vhost_name }}
-        protocol: amqps
+        protocol: amqp
```
Comment (rimusz) [inline stable/pipelines/values.yaml:505]: Please an option to set `amqp` or `amqps`

File: stable/pipelines/values.yaml [hunk 2]
```diff
@@ -533,7 +533,7 @@ pipelines:
           {{- else if .Values.rabbitmq.enabled }}
           url: amqp://{{ tpl .Release.Name . }}-rabbitmq
           {{- else }}
-          url: amqps://{{ .Values.rabbitmq.msg_hostname }}:{{ .Values.rabbitmq.port }}
+          url: amqp://{{ tpl .Values.rabbitmq.msg_hostname . }}:{{ .Values.rabbitmq.port }}
```
Comment (rimusz) [inline stable/pipelines/values.yaml:536]: same here, please an option to set `amqp` or `amqps`

File: stable/pipelines/values.yaml (hunk 1 above)
Comment (jatindogra) [reply stable/pipelines/values.yaml:505]: I agree this should not be a default, there should be an option to set `amqp` or `amqps`

Comment (eldada) [reply stable/pipelines/values.yaml:505]: Please make it as similar to Xray as possible. We would like to eventually share config.
Also, it might be a good idea to add support for the `global` option here... See https://helm.sh/docs/chart_template_guide/subcharts_and_globals/#global-chart-values

**Comment by rimusz** [comment]
The changes should be documented

**Review by rimusz — CHANGES_REQUESTED** [review]
The changes should be documented

File: stable/pipelines/values.yaml [hunk 3]
```diff
@@ -504,7 +504,7 @@ pipelines:
         defaultExchange: {{ .Values.rabbitmq.root_vhost_exchange_name }}
         amqpVhost: {{ .Values.rabbitmq.build_vhost_name}}
         amqpRootVhost: {{ .Values.rabbitmq.root_vhost_name }}
-        protocol: amqps
+        protocol: {{ .Values.rabbitmq.protocol | default "amqps" }}
```
Comment (danielezer) [inline stable/pipelines/values.yaml:507]: Add this to the values instead of using | default.
This causes redundant duplications

File: stable/pipelines/values.yaml [hunk 4]
```diff
@@ -535,7 +535,7 @@ pipelines:
           {{- else if .Values.rabbitmq.enabled }}
           url: amqp://{{ tpl .Release.Name . }}-rabbitmq
           {{- else }}
-          url: amqps://{{ .Values.rabbitmq.msg_hostname }}:{{ .Values.rabbitmq.port }}
+          url: {{ .Values.rabbitmq.protocol | default "amqps" }}://{{ tpl .Values.rabbitmq.msg_hostname . }}:{{ .Values.rabbitmq.port }}
```
Comment (danielezer) [inline stable/pipelines/values.yaml:538]: Same here

File: stable/pipelines/values.yaml [hunk 5]
```diff
@@ -567,8 +567,6 @@ pipelines:
       redis:
         ip: {{ .Release.Name }}-redis-master
         port: 6379
-        image: "jfrog/pipelines-redis"
```
Comment (danielezer) [inline stable/pipelines/values.yaml:570]: Why was this removed?

**Review by danielezer — CHANGES_REQUESTED** [review]

File: stable/pipelines/values.yaml (hunk 5 above)
Comment (chukka) [reply stable/pipelines/values.yaml:570]: This is not needed .. redundant

File: stable/pipelines/values.yaml (hunk 3 above)
Comment (rahulsadanandan) [reply stable/pipelines/values.yaml:507]: @danielezer , This is when we are using an external rabbitmq. If we removed the default (amqps) which was there before. would that break any existing deployments , if they dont provide the protocol. ?

Comment (danielezer) [reply stable/pipelines/values.yaml:507]: @rahulsadanandan I'm not asking you to remove the protocol. I'm asking that you add `rabbitmq.protocol=amqps` to the values.yaml and that will remove the need to use `| default "amqps"`

Comment (rahulsadanandan) [reply stable/pipelines/values.yaml]: Changed.

File: stable/pipelines/values.yaml (hunk 4 above)
Comment (rahulsadanandan) [reply stable/pipelines/values.yaml]: changed

**Review by danielezer — APPROVED** [review]

**Comment by rimusz** [comment]
Also please move external rabbitmq/redis settings to `global:` to be consistent, as there external `postgres` gets set

**Review by jatindogra — APPROVED** [review]

**Review by rahulsadanandan — APPROVED** [review]

**Review by chukka — APPROVED** [review]

**Comment by chukka** [comment]
> The changes should be documented

@rimusz  Done
